activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -103,6 +103,19 @@ def initialize(connection, logger = nil, config = {}) # :nodoc:
         )
       end
 
+      EXCEPTION_NEVER = { Exception => :never }.freeze # :nodoc:
+      EXCEPTION_IMMEDIATE = { Exception => :immediate }.freeze # :nodoc:
+      private_constant :EXCEPTION_NEVER, :EXCEPTION_IMMEDIATE
+      def with_instrumenter(instrumenter, &block) # :nodoc:
+        Thread.handle_interrupt(EXCEPTION_NEVER) do
```

Comment:
This might be a bit excessive, but I'm worried about a killed thread leaving a connection in a broken state.